File: Beecrowd/3 - Strings/1024 - ~/1239 - Bloggo Shortcuts/C/1239_BloggoShortcuts.c
#include <stdio.h>
#include <stdbool.h>

int main() {
    char s[1001];

    while (fgets(s, sizeof(s), stdin)) {
        bool italic = false, bold = false;

        for (int i = 0; s[i] != '\0'; i++) {
            if (s[i] == '\n') {
                continue;
            }

            if (s[i] == '_') {
                if (!italic) {
                    printf("<i>");
                } else {
                    printf("</i>");
                }

                italic = !italic;
            } else if (s[i] == '*') {
                if (!bold) {
                    printf("<b>");
                } else {
                    printf("</b>");
                }
                
                bold = !bold;
            } else {
                putchar(s[i]);
            }
        }
        
        printf("\n");
    }

    return 0;
}

File: Beecrowd/3 - Strings/1024 - ~/1239 - Bloggo Shortcuts/1239_BloggoShortcuts.py
import sys

for line in sys.stdin:
    s = line.rstrip("\n")
    italic = False
    bold = False

    for char in s:
        if char == "_":
            if not italic:
                print("<i>", end="")
            else:
                print("</i>", end="")

            italic = not italic
        elif char == "*":
            if not bold:
                print("<b>", end="")
            else:
                print("</b>", end="")

            bold = not bold
        else:
            print(char, end="")

    print()
